### DISCUSSION TITLE
Installation error

### DISCUSSION THREAD

**Discussion by SimonWetzel** [Xilinx/RapidWright#460] [General]
Hi,
i am trying to install RapidWright on my linux system (Ubuntu 20.04) and I got this error after running 
**$ java -jar rapidwright-installer.jar**:
```

ERROR: Looks like the DeviceBrowser did not run or crashed. Please examine
  the output for clues as to what went wrong.  If you are stumped, please request help
  on the RapidWright Google Group Forum: https://groups.google.com/forum/#!forum/rapidwright.

*** If you are running Linux ***
If you are running Linux, a common problem is to be missing libpng12.so.0.
If you are running a CentOS/RedHat/Fedora distro, try the following:
    sudo yum install libpng12

If you are running a Debian/Ubuntu distro, try the following:
    wget -q -O /tmp/libpng12.deb http://mirrors.kernel.org/ubuntu/pool/main/libp/libpng/libpng12-0_1.2.54-1ubuntu1_amd64.deb && sudo dpkg -i /tmp/libpng12.deb && rm /tmp/libpng12.deb
```

JDK Java and Git was installed before. And I also installed **libpng12.so.0**, but this did not solve the problem.

What is causing this, and how can I fix it?

Thank you!

**Comment by clavin-xlnx** [comment]
Thanks, I tried the previous directions on the RapidWright website and ran into this same error.  I've updated them based on a new package (and installer).  Previously, the recommended instructions on RapidWright's website for Ubuntu on the latest LTS version to install libpng12: 
```
wget -q -O /tmp/libpng12.deb http://mirrors.kernel.org/ubuntu/pool/main/libp/libpng/libpng12-0_1.2.54-1ubuntu1_amd64.deb && sudo dpkg -i /tmp/libpng12.deb && rm /tmp/libpng12.deb
```
failed with:
```
electing previously unselected package libpng12-0:amd64.
dpkg: regarding /tmp/libpng12.deb containing libpng12-0:amd64:
usrmerge conflicts with libpng12-0 (<< 1.2.54-4~)
libpng12-0:amd64 (version 1.2.54-1ubuntu1) is to be installed.

dpkg: error processing archive /tmp/libpng12.deb (--install):
conflicting packages - not installing libpng12-0:amd64
Errors were encountered while processing:
/tmp/libpng12.deb
```
Did you happen to encounter this error?  Attempting the RapidWright installer in this state led to the error you cited above.  I was able to find an updated package because of the conflict it was causing, so re-running it with:
```
wget -q -O /tmp/libpng12.deb https://snapshot.debian.org/archive/debian/20160413T160058Z/pool/main/libp/libpng/libpng12-0_1.2.54-6_amd64.deb && sudo dpkg -i /tmp/libpng12.deb && rm /tmp/libpng12.deb
```
Once I ran this, I could re-run the attempt at the DeviceBrowser with (from the `RapidWright` directory):
```
java -cp bin:jars/* com.xilinx.rapidwright.device.browser.DeviceBrowser
```
And the GUI came up correctly.  If you are still having trouble, try running the DeviceBrowser directly (as in the previous command) and post the error/stack trace here.

**Reply by clavin-xlnx** [reply]
(see #461 for the relevant fix)

**Comment by tutyansed** [comment]
Hi,
i am trying to install RapidWright on my linux system and got the following error while running "**./gradlew compileJava**"

`> Task :run FAILED
Exception in thread "main" java.lang.ExceptionInInitializerError
        at com.xilinx.rapidwright.gui.WidgetMaker.getSupportedDevices(WidgetMaker.java:43)
        at com.xilinx.rapidwright.gui.WidgetMaker.createAvailablePartTreeWidget(WidgetMaker.java:57)
        at com.xilinx.rapidwright.device.browser.DeviceBrowser.initializeSideBar(DeviceBrowser.java:162)
        at com.xilinx.rapidwright.device.browser.DeviceBrowser.<init>(DeviceBrowser.java:111)
        at com.xilinx.rapidwright.device.browser.DeviceBrowser.main(DeviceBrowser.java:96)
        at com.xilinx.rapidwright.MainEntrypoint.main(MainEntrypoint.java:208)
Caused by: java.io.UncheckedIOException: ERROR: Problem downloading file: /wrk/xedhdnobkup2/sedat/rapidWright/RapidWright/data/parts.db, ensure a stable Internet connection.
        at com.xilinx.rapidwright.util.Installer.downloadFile(Installer.java:181)
        at com.xilinx.rapidwright.util.FileTools._downloadDataFile(FileTools.java:919)
        at com.xilinx.rapidwright.util.FileTools.downloadDataFile(FileTools.java:906)
        at com.xilinx.rapidwright.util.FileTools.ensureCorrectDataFile(FileTools.java:940)
        at com.xilinx.rapidwright.util.FileTools.getRapidWrightResourceInputStream(FileTools.java:988)
        at com.xilinx.rapidwright.device.PartNameTools.<clinit>(PartNameTools.java:51)
`

The issues is because **RAPIDWRIGHT_DATA_URL**  fro, "src/com/xilinx/rapidwright/util/FileTools.java" file , isn't available from my site.
Is there a way to resolve this issue 

Thanks,

**Reply by clavin-xlnx** [reply]
@tutyansed - Not sure what is the exact problem, RapidWright attempts to download some necessary data files when it runs.  The error message seems to show a download problem while running the DeviceBrowser, not a Gradle compile.  One way you could try to work around this is to download all of the data files upfront so that RapidWright doesn't have to.  Its a bit more work, but might be worth trying.  You can download these files from the last release (rapidwright_data*.zip):

https://github.com/Xilinx/RapidWright/releases/download/v2022.1.2-beta/rapidwright_data.zip
https://github.com/Xilinx/RapidWright/releases/download/v2022.1.2-beta/rapidwright_data2.zip

Then just unzip them into the `RapidWright` directory.  This should avoid the need to go and download files.

**Reply by tutyansed** [reply]
Thanks!
